### PR TITLE
test and the issue flag in the commit commnad

### PR DESCRIPTION
**Pull request by BishoyHanyRaafat**
@mark-at-pieces  as we discussed I added the test and the issue flag but have some issues with the test I will need help in it

### CODE REVIEW

File: src/pieces/commands/autocommit/autocommit.py [hunk 1]
```diff
@@ -126,40 +122,47 @@ def git_commit(**kwargs):
...
-                issue_number = pos_client.QGPTApi(api_client).relevance(
-                        pos_client.QGPTRelevanceInput(
-                            query=issue_prompt.format(issues=issue_list),
-                            paths=paths,
-                            application=commands_functions.application.id,
-                            model=model,
-                            options=pos_client.QGPTRelevanceInputOptions(question=True)
-                        )).answer.answers.iterable[0].text
-        
+            issue_prompt = """Please provide the issue number that is related to the changes, If nothing related write 'None'.
```
Comment (hal-8999-alpha) [inline src/pieces/commands/autocommit/autocommit.py:127]: Just to verify, you can also manually insert the issue number right? The default we're using is 3.5 turbo I believe. To get consistent results from this at production level we'd have to use function calling from gpt-3.5-turbo-0613. Claude now offers function calling as well.

File: src/pieces/commands/autocommit/autocommit.py [hunk 2]
```diff
@@ -176,32 +179,32 @@ def git_commit(**kwargs):
...
-                    issue_number = issue_number[1:]
-                    validate_issue = False
-                elif issue_number.isdigit():
-                    validate_issue = False
-                elif issue_number == None or issue_number == "":
-                    break    
-            if not validate_issue:
-                commit_message += f" (issue: #{issue_number})"
+        if issue_flag:
+            # Adding the Issue number if the user accept it
```
Comment (hal-8999-alpha) [inline src/pieces/commands/autocommit/autocommit.py:183]: Minor typo in comment: accepts instead of accept

**Review by hal-8999-alpha — CHANGES_REQUESTED** [review]
Overall really good, only pause I have is on the reliability of using the LLM to get the issue number without using function calling. Great work!

**Comment by BishoyHanyRaafat** [comment]
Yeah I think we need to make a todo to split this function for more cleaner and reliable code

File: src/pieces/commands/autocommit/autocommit.py (hunk 1 above)
Comment (hal-8999-alpha) [reply src/pieces/commands/autocommit/autocommit.py:127]: @BishoyHanyRaafat talked with the team, currently Pieces doesn't have the ability for function calling. Would you be opposed to adding an option for the user to manually override whatever chatgpt comes back with and write their own message?

Comment (BishoyHanyRaafat) [reply src/pieces/commands/autocommit/autocommit.py:127]: What we should do to resolve that issue??

**Comment by Bishoy-at-pieces** [comment]
@hal-8999-alpha can we merge this?

**Review by hal-8999-alpha — APPROVED** [review]

**Pull request merged**